net/nut/files/nut-monitor.init

```diff
@@ -2,18 +2,175 @@
+	[ -n "$val" ] && echo "NOTIFYMSG REPLBATT \"$val\"" >> $UPSMON_C
+	config_get val "$cfg" nocommmsg
+	[ -n "$val" ] && echo "NOTIFYMSG NOCOMM \"$val\"" >> $UPSMON_C
+	config_get val "$cfg" noparentmsg
+	[ -n "$val" ] && echo "NOTIFYMSG NOPARENT \"$val\"" >> $UPSMON_C
+
+	notifylist() {
+		local value="$1"
+		append optval "$value" "+"
+	}
+
+	setnotify() {
+		local cfg="$1"
+		local optname="$2"
+		local optval
+		config_list_foreach "$cfg" "$optname" notifylist
+		if [ -z "$optval" ]; then
+			# If no list fallback to trying option, fallback to default
+			config_get optval "$cfg" "$optname" "$default"
+			if [ -n "$optval" ]; then
+				echo "$optval"
+			else
+				# No default, so do the NUT default
+				echo "SYSLOG"
+			fi
+		else
+			echo "$optval"
+		fi
+        }
+
+	local default optval
+	val=""
+	config_list_foreach "$cfg" defaultnotify notifylist
+	default="$optval"
+	echo "NOTIFYFLAG ONLINE $(setnotify "$cfg" onlinenotify)" >> $UPSMON_C
+	echo "NOTIFYFLAG ONBATT $(setnotify "$cfg" onbattnotify)" >> $UPSMON_C
+	echo "NOTIFYFLAG LOWBATT $(setnotify "$cfg" lowbatnotify)" >> $UPSMON_C
+	echo "NOTIFYFLAG FSD $(setnotify "$cfg" fsdnotify)" >> $UPSMON_C
+	echo "NOTIFYFLAG COMMOK $(setnotify "$cfg" commoknotify)" >> $UPSMON_C
+	echo "NOTIFYFLAG COMMBAD $(setnotify "$cfg" commbadnotify)" >> $UPSMON_C
+	echo "NOTIFYFLAG SHUTDOWN $(setnotify "$cfg" shutdownnotify)" >> $UPSMON_C
+	echo "NOTIFYFLAG REPLBATT $(setnotify "$cfg" repolbattnotify)" >> $UPSMON_C
+	echo "NOTIFYFLAG NOCOMM $(setnotify "$cfg" nocommnotify)" >> $UPSMON_C
+	echo "NOTIFYFLAG NOPARENT $(setnotify "$cfg" noparentnotify)" >> $UPSMON_C
+
+	config_get val "$cfg" rbwarntime 43200
+	echo "RBWARNTIME $val" >> $UPSMON_C
+
+	config_get val "$cfg" nocommwarntime 300
+	echo "NOCOMMWARNTIME $val" >> $UPSMON_C
+
+	config_get val "$cfg" finaldelay 5
+	echo "FINALDELAY $val" >> $UPSMON_C
+
+	config_get val "$cfg" certpath
+	if [ -n "$val" ]; then echo "CERTPATH $val" >> $UPSMON_C; fi
+
+	config_get_bool val "$cfg" certverify 0
+	if [ -n "$val" ]; then echo "CERTVERIFY $val" >> $UPSMON_C; fi
+
+	config_get_bool val "$cfg" forcessl 0
+	if [ -n "$val" ]; then echo "FORCESSL $val" >> $UPSMON_C; fi
+}
+
+nut_upsmon_add() {
+	local cfg="$1"
+	local type="$2"
+	local upsname
+	local hostname
+	local port
+	local powervalue
+	local username
+	local password
+	local system
+
+	# If no core config, use defaults
+	[ -r $UPSMON_C ] || nut_upsmon_conf ""
+
+	config_get upsname "$cfg" upsname
+	config_get hostname "$cfg" hostname localhost
+	config_get port "$cfg" port
+	config_get powervalue "$cfg" powervalue 1
+	config_get username "$cfg" username
+	config_get password "$cfg" password
+	system="$upsname@$hostname"
+	if [ -n "$port" ]; then
+	    system="$system:$port";
+	fi
+	echo "MONITOR $system $powervalue $username $password $type" >> $UPSMON_C
 }
 
 start_service() {
-	upsmon -p
+	[ -d $RUN_D ] || mkdir -p $RUN_D
+        mkdir -p /var/etc/nut
+        chmod 640 /var/etc/nut
```

Comment:
The above 2 lines serve the same purposes as the `[ -d /var/run/nut ]` check below?  and they have blankspaces instead of tabs for indentation.